.github/workflows/release-v2.yml

```diff
@@ -20,31 +33,30 @@ jobs:
           release_name: Release ${{ github.ref }}
           draft: false
           prerelease: true
-      - name: Copy release URL into file
-        run: |
-          mkdir release
-          printf "%s" "${{ steps.create_release.outputs.upload_url }}" > release/url.txt
-      - name: Stash file containing the release URL as an artifact
-        uses: actions/upload-artifact@v1
-        with:
-          name: release-url
-          path: ./release
 
+  # builds coordinator, zips stargate-lib folder and uploads the zip to the created release
   build:
-    name: Build
+    name: Build coordinator
+    needs: create-release
     runs-on: ubuntu-latest
     steps:
-      - uses: actions/checkout@v2
+      - uses: actions/checkout@v3
+        ## TODO why do we need a reference here, it would point to the current ref
         with:
           ref: "v2.0.0"
-      - name: Setup Java JDK
-        uses: actions/setup-java@v1
+
+      - name: Set up JDK 8
+        uses: actions/setup-java@v3
         with:
-          java-version: 1.8
+          distribution: 'temurin'
+          java-version: '8'
```

Comment:
We can use either `8` for latest, or specify specific version like `8.0.332`. No strong opinion on how much to pin, although I did have issues with `ccm` failing on specific new version a while ago. But I guess we can pin to specific version if (and only if) JDK-related problems found.